lib/neo4j/active_node/spatial.rb

```diff
@@ -5,38 +5,75 @@ def self.included(other)
-          return spatial_index_name unless index_name
-          # create_index_callback(index_name)
-          @spatial_index_name = index_name
+        attr_reader :spatial_layer_name
+        attr_reader :spatial_layer_type
+        attr_reader :spatial_layer_config
+
+        def spatial_layer(layer_name = nil, options = {})
+          return spatial_layer_name unless layer_name
+
+          @spatial_layer_name = layer_name
+          @spatial_layer_type = options.fetch(:type, 'SimplePoint')
+          @spatial_layer_config = options.fetch(:config, 'lon:lat')
+
+          spatial_layer_name
         end
 
-        # This will not work for now. Neo4j Spatial's REST API doesn't seem to work within transactions.
-        # def create_index_callback(index_name)
-        #   after_create(proc { |node| Neo4j::Session.current.add_node_to_spatial_index(index_name, node) })
-        # end
+        def create_layer
+          fail 'layer name not found' unless spatial_layer_name
 
-        # private :create_index_callback
+          lon_name, lat_name = spatial_layer_config.split(':')
+
+          Neo4j::ActiveBase.current_session.add_layer(spatial_layer_name, spatial_layer_type, lat_name, lon_name)
+        end
+
+        def remove_layer
+          fail 'layer name not found' unless spatial_layer_name
+
+          Neo4j::ActiveBase.current_session.remove_layer(spatial_layer_name)
+        end
       end
     end
 
     module Query
       class QueryProxy
-        def spatial_match(var, params_string, spatial_index = nil)
-          index = model.spatial_index_name || spatial_index
-          fail 'Cannot query without index. Set index in model or as third argument.' unless index
-          Neo4j::Session.current.query
-            .start("#{var} = node:#{index}({spatial_params})")
-            .proxy_as(model, var)
-            .params(spatial_params: params_string)
+        def within_distance(coordinate, distance, layer_name = nil)
+          layer = model.spatial_layer_name || layer_name
+
+          Neo4j::ActiveBase.current_session
+            .within_distance(layer, coordinate, distance, execute: false)
+            .proxy_as(model, :node)
+        end
+
+        def bbox(min, max, layer_name = nil)
+          layer = model.spatial_layer_name || layer_name
+
+          Neo4j::ActiveBase.current_session
+            .bbox(layer, min, max, execute: false)
+            .proxy_as(model, :node)
+        end
+
+        def closest(coordinate, distance = 100, layer_name = nil)
+          layer = model.spatial_layer_name || layer_name
+
+          Neo4j::ActiveBase.current_session
+            .closest(layer, coordinate, distance, execute: false)
+            .proxy_as(model, :node)
+        end
+
+        def intersects(geometry, layer_name = nil)
+          layer = model.spatial_layer_name || layer_name
+
+          Neo4j::ActiveBase.current_session
+            .intersects(layer, geometry, execute: false)
+            .proxy_as(model, :node)
```

Comment:
Again, I'm not familiar with this gem, but using `Neo4j::ActiveBase` and `proxy_as` makes the gem dependent on the `neo4j` gem and thus couldn't be used with the `neo4j-core` gem by itself.  That doesn't seem to be a big deal because `proxy_as` was used previously, but just thought I'd mention it